packages/dd-trace/src/encode/index-0.5.js

```diff
@@ -0,0 +1,144 @@
+'use strict'
+
+const util = require('./util')
+const tokens = require('./tokens')
+const cachedString = require('./cache')(1024)
+
+const MAX_SIZE = 8 * 1024 * 1024 // 8MB
+
+const zeros = {
+  '4': Buffer.alloc(4),
+  '8': Buffer.alloc(8)
+}
+
+let buffer
+let writer
+let traceOffset
+
+function encode (initBuffer, offset, trace, initWriter) {
+  traceOffset = offset
+  buffer = initBuffer
+  writer = initWriter
+  offset = writeArrayPrefix(offset, trace)
+
+  for (const span of trace) {
+    offset = copy(offset, new Uint8Array([0b10011100])) // array of size 12
+    offset = writeUint32(buffer, offset, getStringVal(span.service))
+    offset = writeUint32(buffer, offset, getStringVal(span.name))
+    offset = writeUint32(buffer, offset, getStringVal(span.resource))
+    offset = copy(offset, tokens.uint64)
+    offset = writeId(buffer, offset, span.trace_id)
+    offset = copy(offset, tokens.uint64)
+    offset = writeId(buffer, offset, span.span_id)
+    offset = copy(offset, tokens.uint64)
+    offset = writeId(buffer, offset, span.parent_id)
+    offset = copy(offset, tokens.int64)
+    util.writeInt64(buffer, span.start || 0, offset)
+    offset += 8
+    offset = copy(offset, tokens.int64)
+    util.writeInt64(buffer, span.duration || 0, offset)
+    offset += 8
+    offset = writeInt32(buffer, offset, span.error)
+    offset = writeMap(offset, span.meta || {})
+    offset = writeMap(offset, span.metrics || {})
+    offset = writeUint32(buffer, offset, getStringVal(span.type))
+  }
+
+  return offset
+}
+
+function getStringVal (text = '') {
```

Comment:
I agree. `getNumericEncodedString` is better.

The default value is probably unnecessary given that it operates on presence, rather than type or truthiness, but it was reminding me that in the 0.5 protocol, msgpack `null` is not allowed. That is, string values must always be a string (i.e. if nothing else, there must be an empty string).